packages/starknet-snap/src/rpcs/signTransaction.ts

```diff
@@ -0,0 +1,105 @@
+import type { Component } from '@metamask/snaps-sdk';
+import { heading, UserRejectedRequestError } from '@metamask/snaps-sdk';
+import type { InvocationsSignerDetails } from 'starknet';
+import type { Infer } from 'superstruct';
+import { array, object, string, assign, any } from 'superstruct';
+
+import {
+  confirmDialog,
+  AddressStruct,
+  AuthorizableStruct,
+  BaseRequestStruct,
+  AccountRpcController,
+} from '../utils';
+import { getSignTxnTxt } from '../utils/snapUtils';
+import { signTransactions } from '../utils/starknetUtils';
+
+export const CallStruct = object({
+  contractAddress: string(),
+  calldata: array(any()), // Adjust this as needed for `RawArgs` or `Calldata`
```

Comment:
```suggestion
  calldata: optional(JsonStruct),
```
 
import { JsonStruct } from '@metamask/snaps-utils/dist/json';

move the CallStruct in superstruct as it is share, and have proper test on it, and rename it to TxCallStruct

Reply:
TODO: this struct will be share, will create it from other PR